src/dmd/root/aav.d

```diff
@@ -121,6 +127,63 @@ extern (C++) Value dmd_aaGetRvalue(AA* aa, Key key)
     return null; // not found
 }
 
+debug
+{
+    /**
+    Returns a range of key/values for `aa`.
+    Returns: a range of key/values for `aa`.
+    */
+    @property auto asRange(AA* aa)
+    {
+        return AARange(aa);
+    }
+
+    private struct AARange
+    {
+        AA* aa;
+        size_t b_index;
+        aaA* current;
+
+        this(AA* aa)
+        {
+            if (aa)
+            {
+                this.aa = aa;
+                toNext();
+            }
+        }
+
+        @property bool empty() const { return current is null; }
+
+        @property auto front() { return current.keyValue; }
+
+        void popFront()
+        {
+            if (current.next)
+                current = current.next;
+            else
+            {
+                b_index++;
+                toNext();
+            }
+        }
+
+        private void toNext()
+        {
+            for (; b_index < aa.b_length; b_index++)
+            {
+                auto next = aa.b[b_index];
+                if (next)
```

Comment:
One line: `if (auto next = ...)`

Reply:
actually that logic doesn't work...that assignment can't be done until after the loop condition is satisfied.